components/nexus-core/src/main/java/org/sonatype/nexus/web/CookieFilter.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.web;
+
+import java.util.Collection;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.shiro.web.servlet.AdviceFilter;
+
+import static com.google.common.net.HttpHeaders.SET_COOKIE;
+
+/**
+ * Cookie munging filter.
+ *
+ * @since 2.11.2
+ */
+@Named
+@Singleton
+public class CookieFilter
+    extends AdviceFilter
+{
+  private static final String SECURE_FLAG = "; Secure";
+
+  @Override
+  protected boolean preHandle(final ServletRequest request, final ServletResponse response) throws Exception {
+    filterCookies(request, response);
+    return true;
+  }
+
+  @Override protected void postHandle(final ServletRequest request, final ServletResponse response) throws Exception {
```

Comment:
for some reason IDEA is happy with either annotation on new line or same line :(